preview/MsixCore/msixmgr/msixmgr.cpp

```diff
@@ -137,6 +137,167 @@ int main(int argc, char * argv[])
+                        RETURN_IF_FAILED(currentPackage->GetPackageId(&manifestPackageID));
+
+                        PWSTR packageFullName = nullptr;
+                        RETURN_IF_FAILED(manifestPackageID->GetPackageFullName(&packageFullName));
+
+                        PWSTR packageFileName = nullptr;
+                        RETURN_IF_FAILED(currentPackage->GetFileName(&packageFileName));
+
+                        std::wstring stdPackageFileName = packageFileName;
+                        std::wstring stdPackageFullName = packageFullName;
+
+                        packagesMap[stdPackageFileName] = stdPackageFullName;
+                    }
+
+                    // Use GetPayloadPackages to enumerate over the package files that actually got unpacked
+                    // after applicability logic was applied. 
+                    hasCurrent = false;
+                    ComPtr<IAppxFilesEnumerator> packageFilesEnumerator;
+                    RETURN_IF_FAILED(reader->GetPayloadPackages(&packageFilesEnumerator));
+                    RETURN_IF_FAILED(packageFilesEnumerator->GetHasCurrent(&hasCurrent));
+                    while (hasCurrent)
+                    {
+                        ComPtr<IAppxFile> file;
+                        RETURN_IF_FAILED(packageFilesEnumerator->GetCurrent(&file));
+
+                        PWSTR packageFileName = nullptr;
+                        RETURN_IF_FAILED(file->GetName(&packageFileName));
+
+                        std::wstring stdPackageFileName = packageFileName;
+                        auto packageFullName = packagesMap[packageFileName];
+                        std::wstring packageFolderName = unpackDestination + L"\\" + packageFullName;
+                        packageFolders.push_back(packageFolderName);
+
+                        RETURN_IF_FAILED(packageFilesEnumerator->MoveNext(&hasCurrent));
+                    }
+                }
+
+            }
+            else
+            {
+                return E_INVALIDARG;
+            }
+
+            HMODULE applyACLsDll;
+            applyACLsDll = (LoadLibrary(L"applyacls.dll"));
+            if (applyACLsDll == nullptr)
+            {
+                std::wcout << "Failed to load applyacls.dll. Please confirm the dll is next to this exe" << std::endl;
+            }
+
+            typedef HRESULT(STDMETHODCALLTYPE *APPLYACLSTOPACKAGEFOLDER)(PCWSTR folderPath);
+
+            APPLYACLSTOPACKAGEFOLDER ApplyACLsToPackageFolder =
+                reinterpret_cast<APPLYACLSTOPACKAGEFOLDER>
+                (GetProcAddress(applyACLsDll, "ApplyACLsToPackageFolder"));
+
+            // Update this to pass folder names to API that apply ACLs to those folders
+            for (auto folder : packageFolders)
+            {
+                std::wcout << folder << std::endl;
+                RETURN_IF_FAILED(ApplyACLsToPackageFolder(folder.c_str()));
+            }
+
+            FreeLibrary(applyACLsDll);

```

Comment:
>FreeLibrary(applyACLsDll); [](start = 12, length = 26)

you probably want a smart pointer to auto-free on scope exit. As it is the RETURN_IF_FAILED() failure cases above will skip over this.